### PR TITLE
Fix issue of universe not working for data extraction

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
The universe was not working for indicator data filters.
It is the way we can filter data inside indicator data. ex : Get all Male under 20 `{"gender": "male", "age__lt": "20"}` 
Previously indicator data task expected that we pass the universe from the task but now changed the approach to check if the indicator has a universe object and if there is a universe object present add it to filters.

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-332

## How to test it locally
Add universe to indicator
save variable and wait for the task to complete
Check indicator data results to verify

## Changelog
Changed indicator extraction process
Fixed tests accordingly

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out

### Commits

- [x]  commits are clean
- [x]  commit messages are clean

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers
- [x]  black was run locally (as part of the pre-commit hook)

### Testing

- [x]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: tests/datasets/tasks/indicator_data_extraction/test_extraction.py [hunk 1]
```diff
@@ -57,8 +64,10 @@ def test_three_geographies(self, indicator, geography):
             idata = IndicatorData.objects.get(geography=g)
             assert idata.data[0]["geography"] == g.pk
 
-    def test_universe(self, indicator, geography):
-        universe = {"data__gender": "female", "data__age__lt": "17"}
+    def test_universe(self, indicator, geography, universe):
+        # Add universe to indicator
```
Comment (michaelglenister) [inline tests/datasets/tasks/indicator_data_extraction/test_extraction.py:68]: I think it's worth adding a check here to ensure there is no universe in the indicator before being set

**Review by michaelglenister — COMMENTED** [review]
Changes look good, just a small update needed for the test. Thanks

File: tests/datasets/tasks/indicator_data_extraction/test_extraction.py (hunk 1 above)
Comment (goyal1092) [reply tests/datasets/tasks/indicator_data_extraction/test_extraction.py]: updated test to create and use new indicator rather than updating older one

**Review by michaelglenister — APPROVED** [review]
Approved

File: wazimap_ng/datasets/tasks/indicator_data_extraction.py [hunk 2]
```diff
@@ -9,8 +9,15 @@
...
+
+    universe = {}
+    if indicator.universe and isinstance(indicator.universe.filters, dict):
+        universe = {
+            f"data__{key}": value for key, value in indicator.universe.filters.items()
+        }
+
     geography_ids = (
         models.DatasetData.objects.filter(dataset=indicator.dataset)
         .filter(**universe)
```
Comment (jbothma) [inline wazimap_ng/datasets/tasks/indicator_data_extraction.py:23]: It looks like this is re-implementing what we have in wazimap_ng/datasets/models/datasetdata.py

```python
class DatasetDataQuerySet(models.QuerySet):
    def filter_by_universe(self, universe):
        filters = universe.filters
        if filters and isinstance(filters, dict):
            filters = {f"data__{k}": v for k, v in filters.items()}
            return self.filter(**filters)
        else:
            return self
```

That stopped getting called in 

```
commit 496010d1bb0aa9edd6d6f8ae154dc216a220ed1b
Author: Adi Eyal <adi@openup.org.za>
Date:   Tue Mar 16 15:07:18 2021 +0200

    Refactored indicator_data_extract
    
    Instead of calculating groups, this function simply creates new IndicatorData objects with raw data.
    
    This commit breaks old tests which will be fixed in the following commit.
```

As a matter of style, if we don't use the existing code above, how about renaming `universe` to `filters` or `datasetdata_filters`? Because it's not a Universe instance, it's a dict of filter keys and values.

Comment (jbothma) [reply wazimap_ng/datasets/tasks/indicator_data_extraction.py:23]: so it looks like generating and passing the universe just got lost in the refactoring, and there wasn't a test to detect that.

I'm happy with your implementation if we just change the name as suggested, but also delete the old implementation.

**Review by jbothma — APPROVED** [review]
